lib/engine.js

```
@@ -393,52 +393,51 @@ class Engine {
-            input.url = preSignedUrl;
-            transformerInput = new Asset(input, input.path);
-
-        } // asset is an intermediate rendition from previous transformer and is already available in input.path
-        else if(input.path) {
-            transformerInput = new Asset(input, input.path);
+        if (input && !input.url && !input.path) {
+            throw new GenericError('No source file accessible.');
         }
 
-        if (!transformerInput || !transformerInput.path) {
-            throw new GenericError('No source file accessible.');
+        debug(`sourceType: ${input.sourceType}`);
+        if (input.sourceType === TransformerSourceType.URL) {
+            const inputUrl =  input.url || '';
+            if (input.url && validUrl.isHttpsUri(inputUrl)) {
+                debug(`asset url available : dont need to do anything`);
+                transformerInput = new Asset(input);
+            } else if (input.url && isDataUrlAndValid(inputUrl)) {
+                debug(`asset url available as data URI: upload asset to a temp cloud storage and generate a presigned URL`);
+                this[INTERNAL].timers.upload.start();
+                transformerInput = await getSource(input, transformerDirectory.base, true);
+                this[INTERNAL].timers.upload.stop();
+            } else {
+                debug(`asset url not available: generated presigned url`);
+                try {
+                    this[INTERNAL].timers.upload.start();
+                    const preSignedUrl = await this.getPresignedUrl(input.path);
+                    this[INTERNAL].timers.upload.stop();
+                    input.url = preSignedUrl;
+                    transformerInput = new Asset(input, input.path);
+                } catch (error) {
+                    console.log(error);
+                    throw new GenericError(`Fail generating presigned url.`);
+                }
+            }
+        } else { // LOCAL otherwise
```

please add a note,
defaults to assuming the asset is needed locally

or we can explicitely check it is `LOCAL` and throw if its undefined

---

done

---

default to LOCAL as we don't always specific sourceType in test, I am not sure if that is also the case in execution